lib/px/block/pxblock.lua

```diff
@@ -104,6 +113,26 @@ function M.load(px_config)
             return
         end
 
+        -- json response
+        local accept_header = ngx.req.get_headers()["accept"] or ngx.req.get_headers()["content-type"]
+        local isJsonResponse = accept_header and is_accept_header_json(accept_header) and ngx.ctx.px_cookie_origin == "cookie"
```

Comment:
ngx.ctx.px_cookie_origin == "cookie" why not change this to `isMobile` instead of evaluating this couple of places in the code?

Reply:
there is not really a commonplace worth doing so... other then pxblock, only pxtemplate has this test and none of them loads up commonutils